src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -808,8 +808,10 @@ var ReactCompositeComponentMixin = {
   attachRef: function(ref, component) {
     var inst = this.getPublicInstance();
     invariant(inst != null, 'Stateless function components cannot have refs.');
+    var componentInstance = component.getPublicInstance();
+    invariant(componentInstance != null, 'Stateless function components cannot be given refs.');
```

Comment:
I'm a huge fan of having things fail really hard, but the common consensus among the javascript community is that things should attempt to fail softly when possible.  We can just warn and not set/fire the ref in this case, so that seems more appropriate than crashing the app.  The app may still crash elsewhere as a result, but some apps may be able to continue with minimal disruption.